webservice-utils/src/main/java/org/daisy/pipeline/webserviceutils/xml/ScriptXmlWriter.java

```diff
@@ -147,6 +149,29 @@ private void addOptions(Iterable<XProcOptionInfo> options, Element parent) {
 			parent.appendChild(optionElm);
 		}
 	}
+
+        private void setDefault(XProcOptionInfo info,Element element ){
+                
+                String select=info.getSelect();
+                if (Strings.isNullOrEmpty(select)){
+                        return;
+                }
+                char quote=select.charAt(0);
+                //check that those are quotes
+                if (quote!='"' && quote!='\''){
+                        return;//no quote
+                }
+                //starts and ends by quote
+                if (select.charAt(select.length()-1)!=quote){
+                        return;
+                }
+
+                String def=select.substring(1,select.length()-1);
+                //set whatever is in the select
+                if (def.length()>0){
```

Comment:
An option in XProc can be undefined. If there's no select attribute and no value is provided when the script is invoked then the option value will be undefined. The XProc script can check whether a value is defined in XPath expressions using the function `p:value-available('option-name')`. So potentially a script could behave differently depending on whether the option value is the empty string or if it is undefined. We would need a way in the UIs to choose whether or not to provide an option value for instance though.

I kind of doubt we would use this in our scripts though so it's not that useful, but I think it would be more "correct" to drop the if statement.